fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteSumByAssociativeRule.java

```diff
@@ -296,7 +296,23 @@ private ScalarOperator createNewAggFunction(ScalarOperator arg0, ScalarOperator
                 // if arg1 is nullable, we use count(arg1), otherwise use count()
                 List<ScalarOperator> countArguments = Lists.newArrayList();
                 if (arg1.isNullable()) {
-                    countArguments.add(arg1);
+                    ColumnRefOperator newColumnRef;
+
+                    if (arg1.isColumnRef()) {
+                        newColumnRef = (ColumnRefOperator) arg1;
+                        if (!oldPreAggProjections.containsKey(arg1)) {
+                            newPreAggProjections.put((ColumnRefOperator) arg1, arg1);
+                        }
+                    } else if (commonArguments.containsKey(arg1)) {
+                        // if arg1 has been created by the previous rewriting,
+                        // we don't need to create a new ColumnRef
+                        newColumnRef = commonArguments.get(arg1);
```

Comment:
> why needn't put it to newPreAggProjections?

The presence of arg in commonArguments means that it is already in newPreAggProjections, we use commonArguments to avoid adding repeated ColumnRef